types/ember-data/test/belongs-to.ts

```diff
@@ -29,5 +30,6 @@ folder.set('parent', folder);
 folder.set('parent', folder.get('parent'));
 folder.set('parent', store.findRecord('folder', 3));
 
-// $ExpectType Model | null
-folder.belongsTo('parent').value();
+folder.belongsTo('parent').value(); // $ExpectType Model | null
+folder.belongsTo('parentSync').value(); // $ExpectType Model | null
```

Comment:
Can we introduce a test case for a model with no relationships, just to make sure the `never` case collapses down properly?

Reply:
Honestly, I'm not sure about the

> `never` case collapses down properly

if you mean, making sure that it should not be possible to pass any argument into the `belongsTo`, if there are no any relationships defined on a model.. It seems impossible to do from my experience due to the keys filtering issue described above.

As far as I can see, the best we can do now is to turn `belongsTo(`, and `hasMany(` to return `never` in case of invalid relationship key.

Reply:
Honestly I'm not exactly sure about the

> `never` case collapses down properly

do you mean, to make it impossible to pass any key to the `belongsTo(`/`hasMany(` if there are no any relationships defined on a model? If such, from [my experience](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51852#discussion_r603631700) it seems impossible to do currently.

I think the best we can achieve here for now is to turn `belongsTo(`/`hasMany(` to return `never` for invalid keys, but not fail the build unfortunately.